### DISCUSSION TITLE
Telnet for lwip with W5100

### DISCUSSION THREAD

**Discussion by ZinnerC** [earlephilhower/arduino-pico#1766] [Q&A] [answered]
I'm looking for a Telnet integration with lwip and W5100. Is there a library already working with lwip? Did some crude tests using JAndrassy/TelnetStream by "blindly" copying the respective lines into my code. Did not have success yet. Just wanted to check in if somebody else tried this before me and has some more information.

**Comment by earlephilhower** [comment]
I'm not experienced with that library, but if all you need is a bare pipe over TCP, look at the WiFiClient and WiFiServer examples.  They show bidirectional comms using the Stream interface.

**Reply by earlephilhower** [reply]
A quick look seems to indicate that the `WiFiClient` and `WiFiServer` classes already do all that extra library does.  They export `Stream` and `Print` interfaces so you can read/write them using the expected calls.

**Comment by ZinnerC** [comment, marked as answer]
I would like to give a quick update with respect to the TelnetStream library. @JAndrassy was so kind and updated his library. It is now compatible with lwip and the RP2040.
